src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -2678,6 +2679,28 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
             }
             break;
         }
+        case SMART_ACTION_FALL:
+        {
+            for (WorldObject* target : targets)
+                if (Unit* unitTarget = target->ToUnit())
+                    if (unitTarget->IsAlive() && unitTarget->IsInWorld() && !unitTarget->IsUnderWater())
```

Comment:
IsInWorld is redundant here - units not in world don't execute sai actions
there is no reason to disallow fall while dead or underwater if someone wants to do that